### PR TITLE
build script: git ssh/https option, dependency checking, and fallback git remote namespace

### PR DESCRIPTION
**Pull request by whilei**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -117,11 +142,11 @@ case "$1" in
 		;;
 esac
 
-while getopts "bwd" opt; do
+while getopts "b:wd" opt; do
   case $opt in
     b)
-        echo "Build docs"
-        build 'emerald-cli'
+        echo "Build docs: $OPTARG"
```
Comment (splix) [inline build.sh]: It supposed to be called for all projects, one by one, like:
```
build 'emerald-cli'
build 'emerald-js'
build 'sputnik-vm-spec'
build 'go-ethereum'
build 'json-rpc'
```
The goal is to make whole documentation across all projects in one call, and then publish it on website

**Comment by whilei** [comment]
f9ddf53 allows that now:

```shell
# build all whitelisted projects
./build.sh -b

# build all whitelisted projects
./build.sh -b all

# build one whitelisted project
./build.sh -b emerald-cli

# build a few whitelisted projects
./build.sh -b emerald-cli -b go-ethereum
```

**Comment by whilei** [comment]
Oops, issue with `getopts` syntax

**Comment by whilei** [comment]
Fixed with 3641749, but changed slightly to:

```shell
# build all whitelisted projects
./build.sh -B

# build one whitelisted project
./build.sh -b emerald-cli

# build a few whitelisted projects
./build.sh -b emerald-cli -b go-ethereum
```

(since `getopts` doesn't do optional flag args; https://stackoverflow.com/a/11658375/4401322)

```shell
~/dev/etc/docs-website dev % ⟠ ./build.sh -v
Unknown -v
Use:
--https             : enable https git scheme instead of ssh default
 -B                 : build docs for all available projects

	Available projects are: [emerald-cli go-ethereum]


 -b <project>       : build docs for a single project or multiple

	To build for specific single or multiple projects, use '-b project1 -b project2'

 -w                 : build website
 -d                 : deploy

Examples:

	$ ./build.sh --https -b emerald-cli
	$ ./build.sh -b emerald-cli -b go-ethereum
```

**Comment by splix** [comment]
LGTM

**Pull request merged**